<launch>

  <!-- Set use_sim_time true for datasets-->
  <!-- <param name="use_sim_time" value="true" /> -->

  <node name="object_detector" pkg="dl_node" type="dl_pose_node.py" output="screen"   />


  <group ns="stereo">

    <!-- Call stereo_image_proc to undistort and rectify images -->
    <node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc" />

  </group>

  <!-- Now we crop the images to remove black areas resulting from undistortion.
   Both images must be cropped to the same size
   because intrisic matrices must be of the same for both -->

  <!--
  <node pkg="nodelet" type="nodelet"
        args="standalone image_proc/crop_decimate"
        name="left_decimator">
    <param name="x_offset" type="int" value="12" />
    <param name="y_offset" type="int" value="30" />
    <param name="width" type="int" value="642" />
    <param name="height" type="int" value="314" />


    <remap from="camera/image_raw" to="/stereo/left/image_rect_color" />
    <remap from="camera/camera_info" to="/stereo/left/camera_info" />


    <remap from="camera_out/image_raw" to="/stereo/left_crop/image_rect_color" />
    <remap from="camera_out/camera_info" to="/stereo/left_crop/camera_info" />
  </node>

  <node pkg="nodelet" type="nodelet"
        args="standalone image_proc/crop_decimate"
        name="right_decimator">
    <param name="x_offset" type="int" value="12" />
    <param name="y_offset" type="int" value="30" />
    <param name="width" type="int" value="642" />
    <param name="height" type="int" value="314" />


    <remap from="camera/image_raw" to="/stereo/right/image_rect_color" />
    <remap from="camera/camera_info" to="/stereo/right/camera_info" />


    <remap from="camera_out/image_raw" to="/stereo/right_crop/image_rect_color" />
    <remap from="camera_out/camera_info" to="/stereo/right_crop/camera_info" />
  </node> -->



  <!-- static_transform_publisher x y z yaw pitch roll frame_id child_frame_id -->
  <!-- This transform was used set only to test if everything works
       when using something different than the camera as a reference frame.
       If enabled, take care to also fix the transform below. -->
  <!--node pkg="tf2_ros" type="static_transform_publisher" name="camera_broadcaster" args="1 0 0.5 -1.57 0 -1.57 base_link left_camera" /-->

  <!-- S-PTAM pose path publisher. Used for visualization. -->
  <node pkg="ros_utils" type="pose_to_path" name="sptam_path">
    <remap from="pose" to="sptam/robot/pose" />
    <remap from="path" to="sptam/robot/path" />
  </node>

  <!-- Clean S-PTAM parameters -->
  <rosparam command="delete" ns="sptam" />

  <!-- Nodelet Manager -->
  <node pkg="nodelet" type="nodelet" name="nodelet_manager"  args="manager" output="screen" clear_params="true"  />

  <!-- Choose between Nodelet and Node S-PTAM -->
  <!-- <node pkg="sptam" type="sptam_node" name="sptam" output="screen" > -->
  <node pkg="nodelet" type="nodelet" name="sptam" launch-prefix="xterm -e gdb --args" args="load sptam/sptam_nodelet nodelet_manager" output="screen" clear_params="true"  >

    <!-- Read S-PTAM parameters file -->
    <rosparam command="load" file="$(find sptam)/configurationFiles/zed_vga.yaml" />
    <!-- <param name="LoopDetectorVocabulary" value="$(find sptam)/bow_voc/DBoW2/brief_mit_malaga_vocabulary.yml.gz" /> -->

    <param name="use_prediction" value="false" />
    <param name="approximate_sync" value="false" />
    <param name="publish_transform" value="true" />
    <param name="publish_on_fail" value="true" />

    <!--param name="prediction_frame" value="odom" /-->
    <param name="base_frame" value="camera_left" />
    <param name="camera_frame" value="camera_left" />
    <param name="map_frame" value="map" />
    <param name="reference_frame" value="camera_left" />

    <remap from="/stereo/left/image_rect"  to="/stereo/left/image_rect_color" />
    <remap from="/stereo/right/image_rect"  to="/stereo/right/image_rect_color" />
    <remap from="/stereo/left/camera_info"  to="/stereo/left/camera_info" />
    <remap from="/stereo/right/camera_info"  to="/stereo/right/camera_info" />

  </node>

</launch>
